python/demo/helmholtz2D/demo_helmholtz_2d.py

```diff
@@ -47,18 +47,18 @@
 v = TestFunction(V)
 f = Function(V)
 f.interpolate(lambda x: A * k0**2 * np.cos(k0 * x[0]) * np.cos(k0 * x[1]))
+f.x.scatter_forward()
```

Comment:
Is this call needed? I would have expected the dofs for `f` to all be correctly computed because they just depend on `x`, which available on all ranks.

Reply:
You are right. I thought we only did interpolation on local cells, but we also do it on ghost cells.